### ISSUE TITLE
The similarity matrix is all 1?

### ISSUE THREAD

**Issue by httbupt** [craymichael/CNN_LCD#1]
run "python cnn_lcd.py overfeat".  
Why did I get this?
The similarity matrix is all 1.
![simplot_city_overfeat](https://user-images.githubusercontent.com/37688633/41143045-8012f296-6b2a-11e8-9119-c44a63f21b3d.png)

**Comment by craymichael** [comment]
Hi, do you have any other log output from the run? This could arise from a variety of sources (missing dependencies, incorrectly placed files, etc.). Also, how long did the run take to complete? Thanks

**Comment by httbupt** [comment]
Run time of 100 images in 5 minutes
Log file is shown below

![1](https://user-images.githubusercontent.com/37688633/41187899-ccca0572-6be6-11e8-942f-1cdf67e8bafe.png)

**Comment by craymichael** [comment]
Hmm, everything looks alright other than the segmentation fault. After printing the results the only thing that is run is plotting of the PR-curve so I'm unsure why that would cause core to dump. Regarding the similarity matrix, it is possible that the median filtering is the issue; try running with the argument --sweep_median so that other filter sizes are used. Otherwise, this might indicate the Overfeat Python API could be the culprit; expected results from the alternative TensorFlow models in Python 3 with slim.model would confirm such.

**Comment by httbupt** [comment]
Using TensorFlow model in python 3，the accuracy of experimental results is very low。
![precision-recall_curve_inception_v4](https://user-images.githubusercontent.com/37688633/41581230-7711e086-73d0-11e8-87c8-c2e4b5a436af.png)

**Comment by craymichael** [comment]
Yes, this is to be expected with these models; I was not able to obtain the same results as the paper on either dataset. As far as I can tell, I have the experiments set up identically and have implemented their method as described, and additionally have added median filtering to clean up results (usually improves AP). Anyway, it appears that the code is working and the problem lies with the Overfeat model. I would attempt to reinstall it using the GPU version if you can. The model Python API is unfortunately quite slow and I only refrained from using the TensorFlow implementation due to a lack of easily loadable trained ImageNet weights.

**Comment by httbupt** [comment]
Could you let me see the results of the correct experiments you got using overfeat，Three experimental results。

**Comment by craymichael** [comment]
Hmm, I just pulled the Overfeat code and I appear to be getting the same issue; it appears that the output of every image is the same, which is why the matrix is all 1s. I'll look into it further

**Comment by craymichael** [comment]
Hi, I opened an [issue](https://github.com/sermanet/OverFeat/issues/39) with sermanet about this. Their implementation worked when I was running experiments, but it does not appear to work anymore (it's possible a dependency update has broken things).

**Comment by craymichael** [comment]
Closing per source of the issue originating from [OverFeat repo](https://github.com/sermanet/OverFeat/issues/39).
